### PR TITLE
Save CRC32C Netty allocation to trigger ADLER32 and CRC32 reflective initialization

### PR DESCRIPTION
**Pull request by franz1981**
Motivation:

Allocating a new CRC32C instance cause ByteBufChecksum static init to kick-in, performing reflective lookups of Adler32 and Crc32 OpenJDK methods, although not needed

Modification:

Place each of the reflective methods in their own class holders, to let them kick-in only if directly accessed

Result:

Recheability native image analysis and JIT CRC32C usage won't trigger initializing unused classes or native libraries

### CODE REVIEW

File: codec/src/main/java/io/netty/handler/codec/compression/ByteBufChecksum.java [hunk 1]
```diff
@@ -33,14 +33,13 @@
  * byte array ({@link ByteBuf#hasArray()} is {@code true}) or not.
  */
 abstract class ByteBufChecksum implements Checksum {
-    private static final Method ADLER32_UPDATE_METHOD;
-    private static final Method CRC32_UPDATE_METHOD;
-
-    static {
-        // See if we can use fast-path when using ByteBuf that is not heap based as Adler32 and CRC32 added support
-        // for update(ByteBuffer) in JDK8.
```
Comment (normanmaurer) [inline codec/src/main/java/io/netty/handler/codec/compression/ByteBufChecksum.java:41]: can we just duplicate the comment for both static fields so we dont loose why we did this ?

Comment (franz1981) [reply codec/src/main/java/io/netty/handler/codec/compression/ByteBufChecksum.java:41]: yep good idea :+1:

**Comment by franz1981** [comment]
Just a quick note: this one is to prepare to use the `CRC32C` intrinsics from the OpenJDK via reflection, which doesn't need to load the JNI zip lib, differently from CRC32 and Adler32.

**Review by normanmaurer — APPROVED** [review]

**Comment by franz1981** [comment]
@normanmaurer given that both adler and crc32 load zlib I can just use a single holder class wdyt ?

**Comment by franz1981** [comment]
PTAL @normanmaurer

**Comment by normanmaurer** [comment]
Thanks a lot... main is not affected

**Pull request merged**